### DISCUSSION TITLE
keep node names

### DISCUSSION THREAD

**Discussion by vegarringdal** [zeux/meshoptimizer#244] [General]
Hi

From what I understand keeping mesh names will be hard, but is it possible to keep node names?

I tried with this file:
https://github.com/KhronosGroup/glTF-Sample-Models/blob/master/2.0/GearboxAssy/glTF-Binary/GearboxAssy.glb
Using `gltfpack -i GearboxAssy.glb -o test1.glb -cc -kn -km`
But when I use https://sandbox.babylonjs.com/ and look the node names, they are renamed.

Original:

![image](https://user-images.githubusercontent.com/2901416/108895150-84d5ea80-7613-11eb-8e61-d4a5c5ccb2c7.png)

With your amazing meshoptimizer (4.843MB->1,496MB🥇 ) I see new names.

![image](https://user-images.githubusercontent.com/2901416/108895264-a6cf6d00-7613-11eb-8dfa-c435b8cc65f0.png)

**Comment by zeux** [comment]
None of the names you see are present in the original file I believe. `-kn` should preserve the names *when they are present in the original file*. When names are absent, Babylon.JS generates synthetic names - the order of nodes isn't preserved which is why you see them changed.

**Reply by vegarringdal** [reply]
yes, you are right.

Sorry, I though I saw the same id everytime.
This is great, thanks again for making this amazing lib, really hope I get to try it out more now.

![image](https://user-images.githubusercontent.com/2901416/108904000-40038100-761e-11eb-8728-925d97b228fc.png)
